src/o365/spo/commands/listitem/listitem-record-undeclare.ts

```diff
@@ -0,0 +1,305 @@
+          cmd.log('');
+        }
+        if (this.verbose) {
+          cmd.log(`Undeclare list item  as a record in list ${args.options.listId || args.options.listTitle} in site ${args.options.webUrl}...`);
+        }
+        formDigestValue = res['FormDigestValue'];
+        return this.requestObjectIdentity(args.options.webUrl, cmd, formDigestValue, siteAccessToken);
+      }).then((objectIdentity: string): request.RequestPromise => {
+        this.objectIdentity = objectIdentity;
+        const requestBody: any = `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}"
+                                  xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009">
+                                  <Actions><StaticMethod TypeId="{ea8e1356-5910-4e69-bc05-d0c30ed657fc}"
+                                  Name="UndeclareItemAsRecord" Id="53"><Parameters><Parameter ObjectPathId="49" /></Parameters>
+                                  </StaticMethod></Actions><ObjectPaths><Identity Id="49" Name="${this.objectIdentity}:list:${environmentListId}:item:${args.options.id},1" /></ObjectPaths></Request>`
+        const requestOptions: any = {
+          url: `${args.options.webUrl}/_vti_bin/client.svc/ProcessQuery`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${siteAccessToken}`,
+            'Content-Type': 'text/xml',
+            'X-RequestDigest': formDigestValue,
+          }),
+          body: requestBody
+        }
+
+        if (this.debug) {
+          cmd.log('Executing web request...');
+          cmd.log(requestOptions);
+          cmd.log('');
+          cmd.log('Body:');
+          cmd.log(requestBody);
+          cmd.log('');
+        }
+
+        return request.post(requestOptions);
+      })
+      .then((): void => {
+        // REST post call doesn't return anything
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  };
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-u, --webUrl <webUrl>',
+        description: 'URL of the site where the list item should be undeclared as a record'
+      },
+      {
+        option: '-i, --id <id>',
+        description: 'ID of the list item to be undeclared as a record.'
+      },
+      {
+        option: '-l, --listId [listId]',
+        description: 'ID of the list where the list item should be undeclared as a record. Specify listId or listTitle but not both'
+      },
+      {
+        option: '-t, --listTitle [listTitle]',
+        description: 'Title of the list where the list item should be undeclared as a record. Specify listId or listTitle but not both'
+      }
+
+    ];
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public types(): CommandTypes {
+    return {
+      string: [
```

Comment:
Defining `webUrl`, `listId` and `listTitle` as string shouldn't be necessary because they are strings already. Is there any reason for doing it?